### DISCUSSION TITLE
Hotkey support for menu access

### DISCUSSION THREAD

**Discussion by yepyepyep4711** [djyt/cannonball#105] [Ideas]
Hello,

the fact that the menu can only be bound to a single key is an issue for cabs using hotkeys for these things. On my arcade controllers, there are no hard-to-reach buttons. Thus, is quite easy to hit the menu button by mistake mid-game and lose your progress, which is frustrating.

I see several possibilities:
- enable a hotkey. In my case, I'd put COIN/SELECT as a hotkey (when held) and COIN/SELECT + ACCEL would open the menu
- have a confirmation screen on game exit to menu: "do you want to quit your race to access the menu? Your progress will be lost. Y/N"
- make accessing the menu not quit the game, making it possible to a) go back to the game if you accessed the menu by mistake b) make adjustments mid-game, which would be really handy for testing the different settings anyway

I hope one of these (or any other solution) will be possible.

**Comment by djyt** [comment]
Yep, that's definitely a usability tweak I should entertain, given that CannonBall is in use on a wider array of devices than just desktop PCs these days.

**Reply by yepyepyep4711** [reply]
To further feed your thoughts: [this patch](https://github.com/batocera-linux/batocera.linux/blob/master/package/batocera/emulators/cannonball/0003-hotkey.patch) to your code in the batocera distribution already introduces a hotkey, albeit just for exiting.
